insights/parsers/cryptsetup_luksDump.py

```diff
@@ -0,0 +1,175 @@
+        Label:         	(no label)
+        Subsystem:     	(no subsystem)
+        Flags:       	(no flags)
+
+        Data segments:
+          0: crypt
+                offset: 16777216 [bytes]
+                length: (whole device)
+                cipher: aes-xts-plain64
+                sector: 4096 [bytes]
+
+        Keyslots:
+          0: luks2
+                Key:        512 bits
+                Priority:   normal
+                Cipher:     aes-xts-plain64
+                Cipher key: 512 bits
+                PBKDF:      argon2id
+                Time cost:  7
+                Memory:     1048576
+                Threads:    4
+                AF stripes: 4000
+                AF hash:    sha256
+                Area offset:32768 [bytes]
+                Area length:258048 [bytes]
+                Digest ID:  0
+        Tokens:
+          0: systemd-tpm2
+                Keyslot:    2
+        Digests:
+          0: pbkdf2
+                Hash:       sha256
+                Iterations: 129774
+
+    Examples:
+        >>> type(ros_input)
+        <class 'insights.parsers.ros_config.RosConfig'>
+        >>> ros_input.rules[0]['allow_disallow']
+        'disallow'
+        >>> ros_input.rules[0]['hostlist']
+        ['.*']
+        >>> ros_input.rules[0]['operationlist']
+        ['all']
+        >>> ros_input.specs[0].get('state')
+        'mandatory on'
+        >>> ros_input.specs[0].get('metrics')['mem.util.used']
+        []
+        >>> ros_input.specs[0].get('metrics')['kernel.all.cpu.user']
+        []
+        >>> ros_input.specs[0].get('logging_interval')
+        'default'
+
+        >>> type(dump)
+        Out[6]: insights.parsers.cryptsetup_luksDump.LUKS_Dump
+
+        >>> dump.dump["header"]
+        {'Version': '2',
+         'Epoch': '6',
+         'Metadata area': '16384 [bytes]',
+         'Keyslots area': '16744448 [bytes]',
+         'UUID': 'cfbcc942-e06b-4c4a-952f-e9c9b2011c27',
+         'Label': '(no label)',
+         'Subsystem': '(no subsystem)',
+         'Flags': '(no flags)'}
+
+        >>> dump.dump["Keyslots"]["0"]
+        {'Key': '512 bits',
+         'Priority': 'normal',
+         'Cipher': 'aes-xts-plain64',
+         'Cipher key': '512 bits',
+         'PBKDF': 'argon2id',
+         'Time cost': '7',
+         'Memory': '1048576',
+         'Threads': '4',
+         'AF stripes': '4000',
+         'AF hash': 'sha256',
+         'Area offset': '32768 [bytes]',
+         'Area length': '258048 [bytes]',
+         'Digest ID': '0',
+         'type': 'luks2'}
+
+        >>> dump.dump["Tokens"]["0"]["type"]
+        'systemd-tpm2'
+
+
+    Attributes:
+        dump(dict of dicts): A top level dict containing the dictionaries
+            representing the header, data segments, keyslots, digests
+            and tokens.
+
+    """
+    def parse_dump(self, text):
+        self.dump = None
+        header = LUKS_Header(text)
+
+        if header["header"]["Version"] == "1":
```

Comment:
Generally when using parsr, this logic is in the parsr logic using the | symbol. The way you have it setup you're parsing the content twice it looks like.  Example would be something similar to the below.

`Doc = LUKS_Header + (LUKS1Body | LUKS2Body)`